arrow-core/src/main/kotlin/arrow/core/extensions/list/monad/ListKMonad.kt

```diff
@@ -0,0 +1,283 @@
+  "UNUSED_PARAMETER"
+)
+@Deprecated("@extension projected functions are deprecated", ReplaceWith("flatMap { a -> arg1.value().map { a } }"))
+fun <A, B> List<A>.productLEval(arg1: Eval<Kind<ForListK, B>>): List<A> =
+    arrow.core.extensions.list.monad.List.monad().run {
+  arrow.core.ListK(this@productLEval).productLEval<A, B>(arg1) as kotlin.collections.List<A>
+}
+
+@JvmName("forEffectEval")
+@Suppress(
+  "UNCHECKED_CAST",
+  "USELESS_CAST",
+  "EXTENSION_SHADOWED_BY_MEMBER",
+  "UNUSED_PARAMETER"
+)
+@Deprecated("@extension projected functions are deprecated", ReplaceWith("flatMap { a -> arg1.value().map { a } }"))
+fun <A, B> List<A>.forEffectEval(arg1: Eval<Kind<ForListK, B>>): List<A> =
+    arrow.core.extensions.list.monad.List.monad().run {
+  arrow.core.ListK(this@forEffectEval).forEffectEval<A, B>(arg1) as kotlin.collections.List<A>
+}
+
+@JvmName("mproduct")
+@Suppress(
+  "UNCHECKED_CAST",
+  "USELESS_CAST",
+  "EXTENSION_SHADOWED_BY_MEMBER",
+  "UNUSED_PARAMETER"
+)
+@Deprecated("@extension projected functions are deprecated", ReplaceWith("flatMap { a -> arg1(a).map { b -> Tuple2(a, b) } }"))
+fun <A, B> List<A>.mproduct(arg1: Function1<A, Kind<ForListK, B>>): List<Tuple2<A, B>> =
+    arrow.core.extensions.list.monad.List.monad().run {
+  arrow.core.ListK(this@mproduct).mproduct<A, B>(arg1) as
+    kotlin.collections.List<arrow.core.Tuple2<A, B>>
+}
+
+@JvmName("ifM")
+@Suppress(
+  "UNCHECKED_CAST",
+  "USELESS_CAST",
+  "EXTENSION_SHADOWED_BY_MEMBER",
+  "UNUSED_PARAMETER"
+)
+@Deprecated("@extension projected functions are deprecated", ReplaceWith("ifM(arg1, arg2)", "arrow.core.ifM"))
+fun <B> List<Boolean>.ifM(arg1: Function0<Kind<ForListK, B>>, arg2: Function0<Kind<ForListK, B>>):
+    List<B> = arrow.core.extensions.list.monad.List.monad().run {
+  arrow.core.ListK(this@ifM).ifM<B>(arg1, arg2) as kotlin.collections.List<B>
+}
+
+@JvmName("selectM")
+@Suppress(
+  "UNCHECKED_CAST",
+  "USELESS_CAST",
+  "EXTENSION_SHADOWED_BY_MEMBER",
+  "UNUSED_PARAMETER"
+)
+@Deprecated("@extension projected functions are deprecated", ReplaceWith("selectM(arg1)", "arrow.core.selectM"))
+fun <A, B> List<Either<A, B>>.selectM(arg1: List<Function1<A, B>>): List<B> =
+    arrow.core.extensions.list.monad.List.monad().run {
+  arrow.core.ListK(this@selectM).selectM<A, B>(arrow.core.ListK(arg1)) as kotlin.collections.List<B>
+}
+
+@JvmName("select")
+@Suppress(
+  "UNCHECKED_CAST",
+  "USELESS_CAST",
+  "EXTENSION_SHADOWED_BY_MEMBER",
+  "UNUSED_PARAMETER"
+)
+@Deprecated("@extension projected functions are deprecated", ReplaceWith("select(arg1)", "arrow.core.select"))
+fun <A, B> List<Either<A, B>>.select(arg1: List<Function1<A, B>>): List<B> =
+    arrow.core.extensions.list.monad.List.monad().run {
+  arrow.core.ListK(this@select).select<A, B>(arrow.core.ListK(arg1)) as kotlin.collections.List<B>
+}
+
+/**
+ * cached extension
+ */
+@PublishedApi()
+internal val monad_singleton: ListKMonad = object : arrow.core.extensions.ListKMonad {}
```

Comment:
Does this needs to be deprecated, too?

Reply:
No, it's `internal` to Arrow so no need to deprecate it. It's only used in the same file.